File: cinebook-frontend/src/services/TheaterService.js
import axiosInstance from "./axiosInstance";

export const allTheatres=()=>{
    return axiosInstance.get("/api/theater/all");
}

export const getTheatreById=(id)=>{
    return axiosInstance.get(`/api/theater/${id}`);
}

export const addTheatre=(theatre)=>{
    return axiosInstance.post("/api/theater",theatre);
}

export const updateTheater=(id,theatre)=>{
    return axiosInstance.put(`/api/theater/${id}`,theatre);
}

export const deactivateTheater=(id)=>{
    return axiosInstance.put(`/api/theater/${id}/deactivate`);
}

export const activateTheater=(id)=>{
    return axiosInstance.put(`/api/theater/${id}/activate`);
}

export const searchTheatreByName=(name)=>{
    return axiosInstance.get(`/api/theater/search/${name}`);
}
export const getTheatreByCity=(city)=>{
    return axiosInstance.get(`/api/theater/city/${city}`);
}

export const getTheatreByState=(state)=>{
    return axiosInstance.get(`/api/theater/state/${state}`);
}

export const getActiveTheatre=()=>{
    return axiosInstance.get('/api/theater/running');
}

export const getTheatreByPincode=(pincode)=>{
    return axiosInstance.get(`/api/theater/pincode/${pincode}`);
}
export const getTheatreByMinimumScreens=(totalScreen)=>{
    return axiosInstance.get(`/api/theater/screens/${totalScreen}`);
}

export const countActiveTheaters=()=>{
    return axiosInstance.get("/api/theater/count");
}

File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/TheaterService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.request.TheaterRequestDto;
import com.cinebook.cinebook.dto.response.MovieResponseDto;
import com.cinebook.cinebook.dto.response.TheaterResponseDto;
import com.cinebook.cinebook.entity.Theater;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TheaterService {
    //add Theatre
    TheaterResponseDto addTheatre(TheaterRequestDto requestDto);

    //Get All Theatre
    Page<TheaterResponseDto> getAllTheatre(int size,int page,String sortDir,String sortBy);

    List<TheaterResponseDto>AllTheatres();

    //Get Theatre By id
    TheaterResponseDto getTheatregetById(Long id);

    //Update Theatre
    TheaterResponseDto updateTheatre(Long id,TheaterRequestDto requestDto);
    //Delete Theatre
    void deleteTheatre(Long id);
    //activate theatre
     TheaterResponseDto activateTheater(Long id);

//search theatre
    List<TheaterResponseDto>searchTheatre(String name);
    //search by city
    List<TheaterResponseDto>getTheatreByCity(String city);
    //search by state
    List<TheaterResponseDto>getTheatreByState(String state);
    //only the active theatre
    List<TheaterResponseDto>getTheatre();
    //search by pincode
    List<TheaterResponseDto>getTheatreByPincode(String pincode);

    //get total screen according to theatre
    List<TheaterResponseDto>getTheatreByMinimumScreens(int totalScreen);
 //count active theaters
    long countActiveTheaters();

}
